src/Steps/Release/MergeUpRelease.php

```diff
@@ -0,0 +1,181 @@
+
+        // Merge up to minor branch, then to major branch
+        $this->mergeUp($output, $releasePlanNode, $releaseBranch, $minorBranch, false);
+        $this->mergeUp($output, $releasePlanNode, $minorBranch, $majorBranch, true);
+
+        // Always check out the release branch at the end
+        $library->checkout($output, $releaseBranch);
+    }
+
+    /**
+     * Merge up from one branch to another, setting the appropriate constraints for released dependencies as we go.
+     */
+    protected function mergeUp(
+        OutputInterface $output,
+        LibraryRelease $releasePlanNode,
+        string $mergeFrom,
+        string $mergeInto,
+        string $isMajorBranch
+    ) {
+        $library = $releasePlanNode->getLibrary();
+        $name = $library->getName();
+
+        // Step 1. Checkout branch to merge into
+        $library->checkout($output, $mergeInto);
+
+        $this->log(
+            $output,
+            "Merging library <info>{$name}</info> from <info>{$mergeFrom}</info> to <info>{$mergeInto}</info>"
+        );
+
+        // Step 2. Merge up if there's anything to merge
+        if ($library->hasDiff($output, $mergeFrom)) {
+            // Note: If there are conflicts or other issues merging,
+            // a Gitonomy\Git\Exception\ProcessException will be thrown and cow will exit
+            $library->merge($output, $mergeFrom);
+        } else {
+            $this->log($output, 'Nothing to merge');
+        }
+
+        // Step 3. Convert constraints to the appropriate dev format
+        $this->destabiliseConstraints($output, $releasePlanNode, $isMajorBranch);
+
+        // Step 4. Push branch
+        $library->pushTo('origin');
+    }
+
+    /**
+     * Set constraints for released dependencies to "1.2.x-dev" or "1.x-dev" format.
+     */
+    protected function destabiliseConstraints(
+        OutputInterface $output,
+        LibraryRelease $releasePlanNode,
+        string $isMajorBranch
+    ) {
+        $parentLibrary = $releasePlanNode->getLibrary();
+        $originalData = $composerData = $parentLibrary->getComposerData();
+
+        // Rewrite all dependencies.
+        // Note: only rewrite dependencies for anything that was included in the release.
+        // This mirrors functionality in PublishRelease::stabiliseRequirements()
+        $items = $this->getReleasePlan()->getAllItems();
+        foreach ($items as $item) {
+            $childName = $item->getLibrary()->getName();
+
+            // Ensure this library is allowed to release this dependency (even if shared)
+            if (!isset($composerData['require'][$childName]) || !$parentLibrary->isChildLibrary($childName)) {
+                continue;
+            }
+
+            $version = $item->getVersion();
+            $target = $version->getMajor();
+            if (!$isMajorBranch) {
+                $target .= '.' . $version->getMinor();
+            }
+            $constraint =  $target . '.x-dev';
```

Comment:
What happens when we have dual support constraints such as graphql 3 and 4?  Yes it's a "temporary" hack, though we may also put in similar hacks in the future, or simply we forced in to dual support, and fail to update this here.  I think it's safer to use something like
```php
$contraint = str_replace('@stable', '', $currentConstraint);
$contraint = preg_replace('#([0-9]+\.[0-9]+)\.[0-9]+#', '$1', $constraint) . '.x-dev';
```

Reply:
> Yes it's a "temporary" hack

I don't want to bake solutions for temporary hacks into this. IF we're going to provide functionality to support dual constraints it needs to be across the board and the graphql hack needs to be done properly and merged in.
Otherwise I don't see the point in making some special handling here that doesn't actually matter because cow would never get to this point because the plan can't support dual constraints and the stabalise constraint functionality also can't handle dual constraints.

Reply:
We'll need to retain the "hack", CMS 4 will always retain dual support for graphql, and we need to be retain the ability to release 4.14 in the future if something special comes up.

I guess raise a new PR for whatever hacks were used in the 4.12 beta release?

Reply:
I've reopened https://github.com/silverstripe/cow/issues/207 to handle that. It's beyond the concerns of this PR.